### PR TITLE
Update TFLite concrete function conversion codes to have the associated

### PR DESCRIPTION
**Pull request by abattery**
# Description

Brief Description of the PR:

Providing trackable objects is now recommended for the users. This makes the
concrete function conversion API be based on the new SavedModel importer, which
will enable new TensorFlow Lite features including variable support, resources
and variant tensor, and signature concept.

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [V] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [V ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_addons/seq2seq/tests/decoder_test.py [hunk 1]
```diff
@@ -113,7 +113,8 @@ def _decode(start_tokens, end_token):
     concrete_function = _decode.get_concrete_function(
         tf.TensorSpec([1], dtype=tf.int32), tf.TensorSpec([], dtype=tf.int32)
     )
-    converter = tf.lite.TFLiteConverter.from_concrete_functions([concrete_function])
+    converter = tf.lite.TFLiteConverter.from_concrete_functions(
```
Comment (bhack) [inline tensorflow_addons/seq2seq/tests/decoder_test.py:116]: You need something like:
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/test_utils.py#L29

Comment (bhack) [reply tensorflow_addons/seq2seq/tests/decoder_test.py:116]: You need also to fix black linting

**Review by bhack — CHANGES_REQUESTED** [review]

**Comment by abattery** [comment]
Please take another look :)

File: tensorflow_addons/seq2seq/tests/decoder_test.py (hunk 1 above)
Comment (bhack) [reply tensorflow_addons/seq2seq/tests/decoder_test.py]: As the public API changed you need to wrap this alternative implementation with a condition similar to:
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/test_utils.py#L29

Generally we maintain a compatibility with the latest 3 TF releases (>= 2.5.x).

**Comment by abattery** [comment]
Checks are passing now.

**Review by bhack — APPROVED** [review]

**Review by qlzh727 — APPROVED** [review]

**Pull request merged**